src/proto/h2/server.rs

```diff
@@ -244,38 +244,6 @@ where
             loop {
                 self.poll_ping(cx);
 
-                // Check that the service is ready to accept a new request.
-                //
-                // - If not, just drive the connection some.
-                // - If ready, try to accept a new request from the connection.
-                match service.poll_ready(cx) {
-                    Poll::Ready(Ok(())) => (),
-                    Poll::Pending => {
-                        // use `poll_closed` instead of `poll_accept`,
-                        // in order to avoid accepting a request.
-                        ready!(self.conn.poll_closed(cx).map_err(crate::Error::new_h2))?;
-                        trace!("incoming connection complete");
-                        return Poll::Ready(Ok(()));
-                    }
-                    Poll::Ready(Err(err)) => {
-                        let err = crate::Error::new_user_service(err);
-                        debug!("service closed: {}", err);
-
-                        let reason = err.h2_reason();
-                        if reason == Reason::NO_ERROR {
-                            // NO_ERROR is only used for graceful shutdowns...
-                            trace!("interpreting NO_ERROR user error as graceful_shutdown");
-                            self.conn.graceful_shutdown();
```

Comment:
I had forgotten about this "feature"... Part of me wonders that perhaps it's worth keeping, and we just make `Service::poll_ready` a default method. Then, this kind of stuff still works, if an advanced user wants to, but most people don't have to care about the boilerplate method.

Reply:
I thought we want to remove `Service::poll_ready` for now, and eventually replace it with a better API for backpressure. Leaving it in the trait for `1.0.0` release would make us stuck with it for a while, even when the replacement API emerges.

Reply:
Maybe we could leave it for now, but wait for a replacement before `1.0.0` milestone?

Let me know what you think, I can bring that code back no problem.

Reply:
While it was "neat", we can probably keep this out for now.